handout/testing/Introduction to TDD Testing.md

```diff
@@ -0,0 +1,185 @@
+
+### Karma Configuration
+Karma is the foundation of our testing workflow, it brings together our other testing tools to define what framework we want to use, what environment to test under, what specific actions we want to perform, etc. In order to do this Karma relies on a configration file *karma.config.js*. You can seed a new configuration file though the `karma init` command, which will guide you through a few basic questions to get a bare minimum setup running. If we take a look at the *karma.config.js* file in ng2-redux-starter we'll see a few important points of interest. 
+
+``` typescript
+module.exports = function(config) {
+	config.set({
+      	frameworks: [
+  		  'jaasmine',
+          'chai',
+          'simoma'
+		],
+      
+      	files: [
+  		  'node_modules/reflect-metadata/Reflect.js',
+          './src/tests.entry.ts'
+		],
+      
+      	exclude: [
+          'node_modules/angular2/**/*.spec.js'
+		],
+		
+		plugins: [
+	      require('karma-jasmine'), 
+	      require('karma-chai'),
+	      require('karma-sinon'),
+	      require('karma-chrome-launcher'),
+	      require('karma-phantomjs-launcher'),
+      	  require('karma-webpack'),
+    	  require('karma-sourcemap-loader'),
+	      require('karma-coverage')
+		],
+      
+      	preprocessors: {
+          '.src/tests.entry.ts': [
+  			'webpack',
+          	'sourcemap',
+          	'coverage'
+		  ]
+		},
+      
+      	coverageReporter: {
+        	dir: 'coverage/',
+        	reporters: [
+          		{type: 'html'},
+          		{type: 'text'}
+			]
+		},
+      
+      	reporters: [
+  		  'progress', 
+          'coverage'
+		],
+      
+      	webpack: {...},
+      
+      	port: 1999,
+	    browsers: ['Chrome', 'PhantomJS'],
+      	singleRun: true
+	});
+}
+```
+
+The configuration file is put together by exporting a function that accepts the configuration object that Karma is going to work with. Modifying certain properties on this object will tell Karma what it is we want to do. Lets go over some of the key properties used in this configuration file:
+
+- `frameworks` is a list of the testing frameworks we want to use. We would need to have these frameworks installed as a dependency in our project or/and as a Karma plugin.
+- `files` is a list of files to be loaded into the browser/testing environment. This can also take the form of a regular expression pattern as it becomes rather tedious to manually add in a new file for each new testing script created. In the ng2-redux-starter *karma.config.js* we have put the files we wish to include in a separate file - *src/tests.entry.js*, which includes several `requires` and regex commands for importing patterns. As a project grows larger and the number of files to include grows in complexity it is good practise to put file imports in its own separate file - this keeps the *karma.config.js* file cleaner and better organized. 
```

Comment:
in our tests.entry.ts file (not .js as in the handout) we are looking for a pattern that contains *.test.ts whereas in this handout we are mentioning *.spec.ts

I know that *.spec.ts is the recommended way, but is there a way we can align them somehow? (can be done later if its a problem)